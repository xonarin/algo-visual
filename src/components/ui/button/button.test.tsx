import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

const textProps = 'Я текст';

describe('Тестим состояние кнопки', () => {
    it('Тест кнопки с текстом', () => {
        const tree = renderer.create(<Button text={textProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Тест кнопки без текста', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Тест кнопки в состояние disabled', () => {
        const tree = renderer.create(<Button disabled />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Тест кнопки с индикацией загрузки true', () => {
        const tree = renderer.create(<Button isLoader={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Тест кнопки с индикацией загрузки false', () => {
        const tree = renderer.create(<Button isLoader={false} />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Тест эмуляции клика по кнопке', () => {
        const testClick = jest.fn();
    
        // Рендерим компонент
        render(<Button text={textProps} onClick={testClick} />)
    
        // Находим элемент ссылки
        const button = screen.getByText(textProps);
    
        // Имитируем нажатие на ссылку
        fireEvent.click(button);
            
        // Проверяем, что alert сработал с правильным текстом предупреждения
        expect(testClick).toHaveBeenCalled();
    }); 
})