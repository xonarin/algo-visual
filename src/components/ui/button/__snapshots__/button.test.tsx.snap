// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестим состояние кнопки Тест кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестим состояние кнопки Тест кнопки в состояние disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестим состояние кнопки Тест кнопки с индикацией загрузки false 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестим состояние кнопки Тест кнопки с индикацией загрузки true 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестим состояние кнопки Тест кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Я текст
  </p>
</button>
`;
